lib/ronin/recon/builtin/api/security_trails.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+#
+# ronin-recon - A micro-framework and tool for performing reconnaissance.
+#
+# Copyright (c) 2023-2024 Hal Brodigan (postmodern.mod3@gmail.com)
+#
+# ronin-recon is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ronin-recon is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with ronin-recon.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+require 'ronin/recon/worker'
+require 'ronin/support/text/patterns/network'
+
+require 'async/http/internet/instance'
+require 'set'
+
+module Ronin
+  module Recon
+    module API
+      #
+      # A recon worker that queries https://securitytrails.com and returns subdomains
+      # for a given domain.
+      #
+      class SecurityTrails < Worker
+
+        register 'api/security_trails'
+
+        author "Nicolò Rebughini", email: "nicolo.rebughini@gmail.com"
+        summary "Queries the Domains https://securitytrails.com API"
+        description <<~DESC
+          Queries the Domains https://securitytrails.com API and returns the subdomains
+          of the domain.
+        DESC
+
+        accepts Domain
+        outputs Domain
```

Comment:
Should be `Host` since we're querying the subdomains of the domain.

Reply:
Amended